### PR TITLE
[FLINK-6881] [FLINK-6896] [table] Creating a table from a POJO and defining a time attribute fails

### PR DESCRIPTION
**Pull request by twalthr**
This PR fixes several issues with POJOs and time attributes.

### CODE REVIEW

File: flink-libraries/flink-table/src/main/scala/org/apache/flink/table/api/StreamTableEnvironment.scala [hunk 1]
```diff
@@ -437,39 +437,64 @@ abstract class StreamTableEnvironment(
...
-          // check type of field that is replaced
-          if (idx < fieldTypes.length &&
-            !(TypeCheckUtils.isLong(fieldTypes(idx)) ||
-              TypeCheckUtils.isTimePoint(fieldTypes(idx)))) {
-            throw new TableException(
-              "The rowtime attribute can only be replace a field with a valid time type, such as " +
-                "Timestamp or Long.")
-          }
-          rowtime = Some(idx, name)
+            s"The rowtime attribute can only be replace a field with a valid time type, " +
```
Comment (fhueske) [inline flink-libraries/flink-table/src/main/scala/org/apache/flink/table/api/StreamTableEnvironment.scala]: remove "be" -> `"The rowtime attribute can only replace a field with ..."`

File: flink-libraries/flink-table/src/main/scala/org/apache/flink/table/api/StreamTableEnvironment.scala [hunk 2]
```diff
@@ -437,39 +437,64 @@ abstract class StreamTableEnvironment(
...
-      case (RowtimeAttribute(reference@UnresolvedFieldReference(name)), idx) =>
-        if (rowtime.isDefined) {
+    def extractRowtime(idx: Int, name: String, origName: Option[String]): Unit = {
+      if (rowtime.isDefined) {
+        throw new TableException(
+          "The rowtime attribute can only be defined once in a table schema.")
+      } else {
+        val mappedIdx = streamType match {
+          case pti: PojoTypeInfo[_] =>
+            pti.getFieldIndex(origName.getOrElse(name))
```
Comment (twalthr) [inline flink-libraries/flink-table/src/main/scala/org/apache/flink/table/api/StreamTableEnvironment.scala]: Thanks @sunjincheng121. Good point.

**Comment by twalthr** [comment]
I will merge this...

File: flink-libraries/flink-table/src/main/scala/org/apache/flink/table/codegen/CodeGenerator.scala [hunk 3]
```diff
@@ -1666,14 +1666,8 @@ class CodeGenerator(
     : GeneratedExpression = {
```
Comment (wuchong) [inline flink-libraries/flink-table/src/main/scala/org/apache/flink/table/codegen/CodeGenerator.scala]: The `fieldMapping` is never used, should we remove it?

**Comment by sunjincheng121** [comment]
+1

File: flink-libraries/flink-table/src/main/scala/org/apache/flink/table/codegen/CodeGenerator.scala (hunk 3 above)
Comment (twalthr) [inline flink-libraries/flink-table/src/main/scala/org/apache/flink/table/codegen/CodeGenerator.scala]: We could. Maybe I have to check that again.

**Pull request closed without merging**